src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -1377,68 +1376,94 @@ impl LinkCollector<'_, '_> {
         }
     }
 
-    /// Check if resolution of inline link's display text and explicit link are same.
-    fn check_redundant_explicit_link(
+    /// Resolve display text if the provided link has separated parts of links.
+    ///
+    /// For example:
+    /// Inline link `[display_text](dest_link)` and reference link `[display_text][reference_link]` has
+    /// separated parts of links.
+    fn resolve_display_text(
         &mut self,
-        explicit_res: &Res,
         explicit_link: &Box<str>,
         display_res_info: ResolutionInfo,
         ori_link: &MarkdownLink,
         diag_info: &DiagnosticInfo<'_>,
     ) {
         // Check if explicit resolution's path is same as resolution of original link's display text path, e.g.
+        //
+        // LinkType::Inline:
+        //
         // [target](target)
         // [`target`](target)
         // [target](path::to::target)
         // [`target`](path::to::target)
+        // [path::to::target](target)
+        // [`path::to::target`](target)
         // [path::to::target](path::to::target)
         // [`path::to::target`](path::to::target)
         //
+        // LinkType::ReferenceUnknown
+        //
+        // [target][target]
+        // [`target`][target]
+        // [target][path::to::target]
+        // [`target`][path::to::target]
+        // [path::to::target][target]
+        // [`path::to::target`][target]
+        // [path::to::target][path::to::target]
+        // [`path::to::target`][path::to::target]
+        //
+        // LinkType::Reference
+        //
+        // [target][target]
+        // [`target`][target]
+        // [target][path::to::target]
+        // [`target`][path::to::target]
+        // [path::to::target][target]
+        // [`path::to::target`][target]
+        // [path::to::target][path::to::target]
+        // [`path::to::target`][path::to::target]
+        //
+        // [target]: target // or [target]: path::to::target
+        // [path::to::target]: path::to::target // or [path::to::target]: target
+        //
```

Comment:
this is quite a long comment. can you move it to the `tests/rustdoc-ui` test you wrote instead? or if it's already there, link to that file instead of duplicating it here?

Reply:
No problem.